src/consumer.ts

```diff
@@ -54,7 +54,10 @@ export class Consumer extends TypedEventEmitter {
   private waitTimeSeconds: number;
   private authenticationErrorTimeout: number;
   private pollingWaitTimeMs: number;
+  private pollingCompleteWaitTimeMs: number;
   private heartbeatInterval: number;
+  private isPolling: boolean;
+  private beganWaitingForStop: number;
```

Comment:
I'm not crazy about the naming or need for this `beganWaitingForStop`. Open to suggestions for alternative

Reply:
Maybe just `stopRequestedAtTimestamp`?

Reply:
I like it - updated!